source/server/server.h

```diff
@@ -293,6 +293,11 @@ class InstanceImpl final : Logger::Loggable<Logger::Id::main>,
 
   Quic::QuicStatNames& quicStatNames() { return quic_stat_names_; }
 
+  void setSinkPredicates(std::unique_ptr<Envoy::Stats::SinkPredicates> sink_predicates) override {
+    sink_predicates_ = std::move(sink_predicates);
+    stats_store_.setSinkPredicates(*sink_predicates_);
```

Comment:
per above I think we should store this in the allocator.

Or is the issue that we don't have a stats allocator at the time this is called?

Reply:
I've stored it in the ThreadLocalStore, as we will need it there for histograms later.

Reply:
could we leave it in the allocator and reference it from thread_local_store? You could in theory make more than one thread_local_store on the same allocator and then you'd have a bit of a pickle :)

Reply:
Fixed.